File: worksheets/dp-five/scala/src/visitor/HtmlTag.scala
package visitor

abstract class HtmlTag extends Element {
  def getTagName(): String
  def setStartTag(tag: String): Unit
  def getStartTag(): String
  def getEndTag(): String
  def setEndTag(tag: String): Unit

  def setTagBody(tagBody: String): Unit = ???
  def addChildTag(htmlTag: HtmlTag): Unit = ???
  def removeChildTag(htmlTag: HtmlTag): Unit = ???
  def getChildren(): List[HtmlTag] = ???
  def generateHtml(): Unit
}

File: worksheets/dp-five/kotlin/src/visitor/HtmlTag.kt
package visitor

abstract class HtmlTag(
        startTag: String,
        endTag: String) : Element {

    abstract fun addChildTag(htmlTag: HtmlTag)
    abstract fun removeChildTag(htmlTag: HtmlTag)
    abstract fun getChildren(): List<HtmlTag>
    abstract fun generateHtml()
}
